jboss/container/java/singleton-jdk/configure.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+set -u
+set -e
+
+# Clean up any java-* packages that have been installed that do not match
+# our stated JAVA_VERSION-JAVA_VENDOR (e.g.: 11-openjdk; 1.8.0-openj9)
```

Comment:
Perhaps there should be a check whether `${JAVA_VERSION}` and `${JAVA_VENDOR}` is empty as the rest of the logic assumes non-empty. It would remove all JDKs in that case, AFAIUI.

Reply:
If they're undefined then the script will fail due to `set -u`. Can they be defined-but-empty? How would we end up in that situation?

Reply:
Assuming it didn't bail out, the grep pattern would be "^java--" which wouldn't match any real java packages.

Reply:
Right, but we have it negated:
```
    if ! echo "$pkg" | grep -q "^java-${JAVA_VERSION}-${JAVA_VENDOR}"; then
```

So for any package, `grep -q "^java--"` won't match, turning that to true.

Reply:
I see what you mean! OK I'll add a pre-check.